### PR TITLE
KA10: Type 340 device is flickering

### PR DESCRIPTION
**Pull request by larsbrinkhoff**
Some 340 applications running in timesharing displays a badly flickering view.  These include FED, SPCWAR, and MacHack VI (the program eloquently named C).

### CODE REVIEW

**Comment by larsbrinkhoff** [comment]
I'm posting a pull request for review and discussion.

I remember from #64 I saw some problems with the device interrupt handling.  My first commit rearranges the interrupt handling to make sure an interrupt is flagged whenever the DONE or SPECIAL status bits are on, and is cleared when they are both off.  Also, the delay for the DONE interrupt is changed so that the bit is not set until the delay has expired.

This is a marked improvement for FED.  The second commit tunes down the delay which makes the display quite stable.

However, SPCWAR and C are still flickering badly.

**Comment by larsbrinkhoff** [comment]
I tested PEEK just now.  Before my change, it was prefectly fine.  But after, it's bad.

File: PDP10/ka10_dpy.c [hunk 1]
```diff
@@ -180,10 +180,24 @@ const char *dpy_description (DEVICE *dptr)
...
+    uptr->DELAYED_DONE = 1;
     uptr->INT_COUNTDOWN = INT_COUNT;
 }
 
+/* update interrupt request */
+static void check_interrupt (UNIT *uptr)
+{
+    if (uptr->STAT_REG & CONI_INT_SPEC) {
+        uint32 sc = uptr->STAT_REG & CONX_SC;
+        set_interrupt(DPY_DEVNUM, sc >> CONX_SC_SHIFT);
```
Comment (philbudne) [inline PDP10/ka10_dpy.c]: Is it OK to call set_interrupt with a zero value? Same thing at line 195.  Would be nice if the comment at 187 explains who calls this.  The name "check_interrupt" sounds a little like this is about interrupt service, when it's about sending interrupts.

Comment (larsbrinkhoff) [reply PDP10/ka10_dpy.c]: Good points.  Yes, that should be ok.  The set_interrupt function checks for zero itself.

**Comment by larsbrinkhoff** [comment]
I'm closing this to make it clear the branch is not in a good state.

**Comment by larsbrinkhoff** [comment]
This shows MacHack flickering quite a bit, at 5:45.  
https://www.csail.mit.edu/node/6666

**Comment by larsbrinkhoff** [comment]
A random thought.  If DISPLAY_CYCLE_US is 50, that means the svc function is called every 50 microseconds, or thereabouts.  On top of that, ka10_dpy.c counts down INT_COUNT times before triggering an interrupt.  The default is (500/DISPLAY_CYCLE_US), so every 0.5 milliseconds.  This goes for data interrupts, not special interrupts. 

I wonder if this rate is right?  Consider point mode.  One of those instructions are documented to take 3 𝜇s for data transfer (an 18-bit instruction?), 35 𝜇s for positioning, and 0.5 𝜇s for intensification.  The total is 38.5 𝜇s, or 77 𝜇s per 36-bit word.

Or consider vector or incremental mode.  The time per point is 3 𝜇s + 1.5 𝜇s.  I think that goes for each point in a vector, so a, say, 200-point vector would be 1.8 ms.  Or a whopping 3.6 ms if there are two such vectors in a PDP-10 word.

I don't see how long it takes to draw a character (but I didn't look too hard).  I think a reasonable guess is to treat is at 5x7 incremental points, so 3 𝜇s + 35x1.5 𝜇s = 55.5 𝜇s.

**Comment by larsbrinkhoff** [comment]
According to a PDP-7 manual, 35 𝜇s.

**Pull request closed without merging**